runtime/pack/dist/opt/cfilter/plugin/cfilter.lua

```diff
@@ -0,0 +1,138 @@
+----------------------------------------
+-- This file is generated via github.com/tjdevries/vim9jit
+-- For any bugs, please first consider reporting there.
+----------------------------------------
+
+-- Ignore "value assigned to a local variable is unused" because
+--  we can't guarantee that local variables will be used by plugins
+-- luacheck: ignore 311
+
+local vim9 = require('_vim9script')
+local M = {}
+local Qf_filter = nil
+-- vim9script
+
+-- # cfilter.vim: Plugin to filter entries from a quickfix/location list
+-- # Last Change: Jun 30, 2022
+-- # Maintainer: Yegappan Lakshmanan (yegappan AT yahoo DOT com)
+-- # Version: 2.0
+-- #
+-- # Commands to filter the quickfix list:
+-- #   :Cfilter[!] /{pat}/
+-- #       Create a new quickfix list from entries matching {pat} in the current
+-- #       quickfix list. Both the file name and the text of the entries are
+-- #       matched against {pat}. If ! is supplied, then entries not matching
+-- #       {pat} are used. The pattern can be optionally enclosed using one of
+-- #       the following characters: ', ", /. If the pattern is empty, then the
+-- #       last used search pattern is used.
+-- #   :Lfilter[!] /{pat}/
+-- #       Same as :Cfilter but operates on the current location list.
+-- #
+
+Qf_filter = function(qf, searchpat, bang)
+  qf = vim9.bool(qf)
+  local Xgetlist = function() end
+  local Xsetlist = function() end
+  local cmd = ''
+  local firstchar = ''
+  local lastchar = ''
+  local pat = ''
+  local title = ''
+  local Cond = function() end
+  local items = {}
+
+  if vim9.bool(qf) then
+    Xgetlist = function(...)
+      return vim.fn['getqflist'](...)
+    end
+    Xsetlist = function(...)
+      return vim.fn['setqflist'](...)
+    end
+    cmd = ':Cfilter' .. bang
+  else
+    Xgetlist = function(...)
+      local copied = vim.deepcopy({ 0 })
+      for _, val in ipairs({ ... }) do
+        table.insert(copied, val)
+      end
+
+      local funcref = 'getloclist'
+      if type(funcref) == 'function' then
+        return funcref(unpack(copied))
+      elseif type(funcref) == 'string' then
+        return vim.fn[funcref](unpack(copied))
+      else
+        error(string.format('unable to call funcref: %s', funcref))
+      end
+    end
+
+    Xsetlist = function(...)
+      local copied = vim.deepcopy({ 0 })
+      for _, val in ipairs({ ... }) do
+        table.insert(copied, val)
+      end
+
+      local funcref = 'setloclist'
+      if type(funcref) == 'function' then
+        return funcref(unpack(copied))
+      elseif type(funcref) == 'string' then
+        return vim.fn[funcref](unpack(copied))
```

Comment:
can this pattern be encapsulated in _vim9script.lua or it must be inlined?

Reply:
yeah, i inlined and fixed a few other bugs while I was testing this (that didn't and still don't apply to this transpilation... found a few bugs in vim9script as well haha)